### PR TITLE
Backport "Fixes for sos-report >= 4.0 versions" to 0.69

### PR DESCRIPTION
**Pull request by ndokos**
Backport of #2372.

### CODE REVIEW

**Comment by ndokos** [comment]
Tested on F33 (sos v4.0) and F34 (sos v4.1)

File: agent/util-scripts/pbench-sysinfo-dump [hunk 1]
```diff
@@ -116,10 +120,18 @@ function collect_sos {
...
+	fi
+	if [[ "${sos_ver}" -gt 4 || ( "${sos_ver}" -eq 4 && "${sos_ver_minor}" -ge 1 )]]; then
+		# sos uses sha256 sums, not md5, in 4.1+
+		_cksum_suffix="sha256"
+	fi
 
 	_name="pbench-$(hostname -f)"
 	_cmd="${dir}/sosreport-${_name}.cmd"
-	printf -- "sosreport" > ${_cmd}
+	printf -- "%s" "${_sos_report_cmd}" > ${_cmd}
```
Comment (webbnh) [inline agent/util-scripts/pbench-sysinfo-dump:134]: While this obviously works, it's more complex than it needs to be; e.g.:
```
	printf -- "${_sos_report_cmd}" > ${_cmd}
```
If you're worried about there being percent signs in the command string, then use something even simpler, like:
```
	echo ${_sos_report_cmd} > ${_cmd}
```

**Review by webbnh — APPROVED** [review]
It looks good except for line 144 (and the nit I called out below).

**Review by dbutenhof — APPROVED** [review]

**Comment by portante** [comment]
Before we consider this back-port let's get the original changes merged into `main` and then revisit.

**Review by portante — CHANGES_REQUESTED** [review]
Let's hold off on this work until we get PR #2372 merged into `main`.

File: agent/util-scripts/pbench-sysinfo-dump (hunk 1 above)
Comment (ndokos) [reply agent/util-scripts/pbench-sysinfo-dump:134]: See my reply in #2372.

**Comment by ndokos** [comment]
Changed the comment on l.144

**Review by portante — APPROVED** [review]

File: agent/util-scripts/pbench-sysinfo-dump [hunk 2]
```diff
@@ -129,13 +141,14 @@ function collect_sos {
 	debug_log "[${script_name}]collecting sosreport"
 	${_cmd} > ${dir}/sosreport-${_name}.log 2>&1
 
-	# The latest version of sosreport (3.6 right now) generates different names
-	# for the sosreport tar balls, inserting a short hostname before the label.
-	# Since older versions of sosreport did not do that, we use a wildcard
-	# pattern to capture both names when checking for success.
+	# Version of sosreport >= 3.6 generate different names for the
```
Comment (webbnh) [inline agent/util-scripts/pbench-sysinfo-dump:144]: If we want this to be perfect, then

s/Version/Versions/

but I'm not going to quibble.

**Review by webbnh — APPROVED** [review]

File: agent/util-scripts/pbench-sysinfo-dump (hunk 2 above)
Comment (ndokos) [reply agent/util-scripts/pbench-sysinfo-dump:144]: Sigh ... I'll leave it as is for now.

**Pull request merged**